pkg/keys/deprecated_ecdsa.go

```diff
@@ -98,6 +116,6 @@ func (p *deprecatedP256Verifier) UnmarshalPublicKey(key *data.PublicKey) error {
 	}
 
 	p.key = key
-	fmt.Fprintln(os.Stderr, "tuf: warning using deprecated ecdsa hex-encoded keys")
+	p.logger.Print(WarnDeprecatedEcdsaKey)
```

Comment:
Maybe gate this behind a https://pkg.go.dev/sync#Once in the struct?

Is "once per verifier" okay? Or is that still too much?

Reply:
TBH I'm not convinced we need to have this at all, if clients opt into it, they probably intend for this to be the case.

Reply:
RIGHT! I went through the same thought. 

I thought: once per client makes sense, but then I thought: this really shouldn't be happening, and if you feel logs are spammy, go log to a file you don't care for

Reply:
Oh, I meant the opposite—do we need to log at all? It shouldn't be a surprise to anybody that this is happening.

Reply:
Oh. I think it's hidden behind 3 layers of deps (cosign -> sigstore -> go-tuf) if not more, so I thought so.

It would be nice for my sigstore layer testing in root-signing to have the affirmation that no longer need that codepath and we can deprecate it. But I'm up for just removing it entirely.

Reply:
I don't feel strongly. But I do think it's kinda gross to expose this to *clients* (e.g. cosign users). I guess Cosign can just swallow it which helps.

Up to you!

Reply:
Yes, the point is that by default it does not log any messages at all, but we can flip the log in, let's say, the root-signing presubmits.

Will think! I have to sign off :/

Reply:
@theupdateframework/go-tuf-maintainers OK: I think I'm going to remove the deprecated ECDSA log message, and leave logging to the repository top-level options.

Are there any client operations that would benefit from info logging? logging updated metadata versions? I went through, and nothing seems like an exceptional case that requires logging (to me). If not, I can just clear out all client logging.

Reply:
IMO logging at the debug level would be really useful for complicated things like delegations. But that can happen later; I'm happy to have it all ripped out now.

Reply:
True! Searching delegations for a method like `GetTarget` would benefit from debug logging which target delegation was required. 

Removing :) someone can take a look at this PR if they want to grab the client logging code then.